docs/autostart_win.rst

```diff
@@ -0,0 +1,105 @@
+.. windows service guide
+
+==================================
+Setting up auto-restart on Windows
+==================================
+
+-------------------------------
+Downloading NSSM via Chocolatey
+-------------------------------
+
+.. note:: This assumes you already have chocolatey installed. 
+          If not, you may install it or follow the manual instructions for getting NSSM.
+
+To install via Chocolatey, search "powershell" in the start menu,
+right-click on it and then click "Run as administrator"
+
+Then run the following command:
+
+.. code-block:: none
+
+    choco install nssm -y
+
+------------------------
+Manually installing NSSM
+------------------------
+
+Go to `<https://nssm.cc/download>`_ and download the latest release. Once the download
+is finished, open File Explorer, navigate to the drive Windows is installed on, and 
+create a folder called :code:`nssm`. Then navigate to your Downloads folder, and move 
+:code:`nssm.exe` to :code:`C:\nssm`.
+
+Then open Powershell as administrator (search "powershell" in the start menu, then right-click and 
+click "Run as administrator") and run the following commands to add NSSM to your path.
+
+.. code-block:: none
+
+    $oldPath = (Get-Itemproperty -path 'hklm:\system\currentcontrolset\control\session manager\environment' -Name Path).Path
+    $newPath = $oldPath + ";C:\nssm"
+    Set-ItemProperty -path 'hklm:\system\currentcontrolset\control\session manager\environment' -Name Path -Value $newPath
+
+Once finished, log out and back in to ensure the changes take effect.
+
+----------------------
+Setting up the service
+----------------------
+
+To create the service, run the following command:
+
+.. note:: If Red was installed in a venv, please activate it before running this command
+
+.. code-block:: none
+
+    nssm install redbot $(Get-Command redbot).Definition <instance name> --no-prompt
+
+replacing <instance name> with the name of the instance.
+
+Then run the following set of commands one at a time:
+
+.. code-block:: none
+
+    nssm set redbot AppExit 0 Exit
+    nssm set redbot AppStopMethodConsole 10000
+    nssm set redbot AppRestartDelay 15000
+    nssm set redbot AppStdout $Env:USERPROFILE\red-stdout.log
+    nssm set redbot AppStderr $Env:USERPROFILE\red-stderr.log
```

Comment:
I think it would be better to use one log file for both stdout and stderr (nssm support that), it's easier to read like this, especially that there are no timestamps in logs (for anyone else not familiar with the issue we had before - enabling `AppTimestampLog` unfortunately breaks auto-restarting and proper service exiting)